src/commands/moderation/prefixCommands/functions/setContent.ts

```diff
@@ -0,0 +1,232 @@
+    }
+    if (!foundCommands || foundCommands.length !== 1) {
+        await interaction.reply({ embeds: [noCommandEmbed(command)], ephemeral: true });
+        return;
+    }
+
+    const foundCommand = foundCommands[0];
+    const { _id: commandId } = foundCommand;
+    let versionId = '';
+    let foundVersions = null;
+    if (version === 'GENERIC' || version === 'generic') {
+        versionId = 'GENERIC';
+    } else {
+        foundVersions = await PrefixCommandVersion.find({ name: version });
+        if (foundVersions && foundVersions.length === 1) {
+            [{ _id: versionId }] = foundVersions;
+        } else {
+            await interaction.reply({ embeds: [noVersionEmbed(version)], ephemeral: true });
+            return;
+        }
+    }
+
+    const foundContent = foundCommand.contents.find((c) => c.versionId.toString() === versionId.toString());
+    const contentModal = new ModalBuilder({
+        customId: 'commandContentModal',
+        title: `Content for ${command} - ${version}`,
+    });
+
+    const commandContentTitle = new TextInputBuilder()
+        .setCustomId('commandContentTitle')
+        .setLabel('Title')
+        .setPlaceholder('Provide a title for the command.')
+        .setStyle(TextInputStyle.Short)
+        .setMaxLength(255)
+        .setMinLength(0)
+        .setRequired(false)
+        .setValue(foundContent ? foundContent.title : '');
+
+    const commandContentContent = new TextInputBuilder()
+        .setCustomId('commandContentContent')
+        .setLabel('Content')
+        .setPlaceholder('Provide the content for the command.')
+        .setStyle(TextInputStyle.Paragraph)
+        .setMaxLength(2048)
+        .setMinLength(0)
+        .setRequired(false)
+        .setValue(foundContent && foundContent.content ? foundContent.content : '');
+
+    const commandContentImageUrl = new TextInputBuilder()
+        .setCustomId('commandContentImageUrl')
+        .setLabel('Image URL')
+        .setPlaceholder('Provide an optional Image URL for the command.')
+        .setStyle(TextInputStyle.Short)
+        .setMaxLength(255)
+        .setMinLength(0)
+        .setRequired(false)
+        .setValue(foundContent && foundContent.image ? foundContent.image : '');
+
+    const titleActionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(commandContentTitle);
+    const contentActionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(commandContentContent);
+    const imageUrlActionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(commandContentImageUrl);
+
+    contentModal.addComponents(titleActionRow);
+    contentModal.addComponents(contentActionRow);
+    contentModal.addComponents(imageUrlActionRow);
+
+    await interaction.showModal(contentModal);
+
+    const filter = (interaction: {
+        customId: string;
+        user: { id: any; };
+    }) => interaction.customId === 'commandContentModal' && interaction.user.id;
```

Comment:
```suggestion
    const filter = (interaction: ModalSubmitInteraction) => interaction.customId === 'commandContentModal' && interaction.user.id;
```
Why not use the actual type?

Reply:
Good suggestion, and it actually indicated a bug! I intended t check that the interaction user was actually the user who executed the command in the first place, which was missing!